tests/test_store.py

```diff
@@ -224,3 +226,37 @@ def test_healthcheck_bad_returns_500(self):
             r = self.app.get(self.endpoints['healthcheck'])
             self.assertEqual(r.status_code, 500)
             self.assertEqual(r.json, {'message': 'Failed to connect to database', 'status': 500})
+
+    def test_delete_old_returns_204_for_no_deletes(self):
+        settings.RESPONSE_RETENTION_DAYS = 90
+        r = self.app.post(self.endpoints['delete-old'])
+        self.assertEqual(r.status_code, 204)
+
+    def test_delete_old_returns_204_when_records_deleted(self):
+        self.app.post(self.endpoints['responses'],
+                      data=second_test_message,
+                      content_type='application/json')
+
+        self.app.post(self.endpoints['responses'],
+                      data=test_message,
+                      content_type='application/json')
+        settings.RESPONSE_RETENTION_DAYS = -2  # Set retention negative so that all added records will get deleted
+        r = self.app.post(self.endpoints['delete-old'])
+        self.assertEqual(r.status_code, 204)
+
+    def test_delete_old_returns_500_if_not_set_in_config(self):
+        settings.RESPONSE_RETENTION_DAYS = None
+        r = self.app.post(self.endpoints['delete-old'])
+        self.assertEqual(r.status_code, 500)
+
+    def test_delete_old_does_not_delete_records_younger_than_response_retention_days(self):
```

Comment:
This doesnt look like its verifying that records are not deleted. The only assertion is on a 204 which you get when records are both deleted and not deleted giving us a potential false positive.

Reply:
ta, added log checking